ql/src/java/org/apache/hadoop/hive/ql/parse/UpdateDeleteSemanticAnalyzer.java

```diff
@@ -62,7 +72,9 @@ protected void analyze(ASTNode tree, Table table, ASTNode tabNameNode) throws Se
     switch (tree.getToken().getType()) {
     case HiveParser.TOK_DELETE_FROM:
       operation = Context.Operation.DELETE;
-      reparseAndSuperAnalyze(tree, table, tabNameNode);
+      if (!tryMetadataUpdate(tree, table, tabNameNode)) {
```

Comment:
Q: could we apply this optimization in update and merge operations as well?

Reply:
Not right now. We are only doing it for delete operations. Need to explore if this is even possible.

Reply:
how about `merge-delete`, however, we might cover this in a follow up jira

Reply:
If its converted to a delete query itself then I think it should work. But as far as I see, I didnt see any merge delete being updated in any QFiles.

Reply:
merge is handled by MergeSemanticAnalyzer (COW) or SplitMergeSemanticAnalyzer, so it's not covered by existing code

Reply:
Yes, I can see some improvements that can be done along this line since there is a clear pathway defined for DELETE clauses when used in merge statement - 
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/parse/MergeSemanticAnalyzer.java#L204-L212